code/game/machinery/computer/communications.dm

```diff
@@ -411,15 +411,35 @@ GLOBAL_VAR_INIT(cops_arrived, FALSE)
 					to_chat(usr, span_warning("You are not authorized to do this!"))
 					playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, FALSE)
 					return
+			else
+				to_chat(usr, "The console refuses to let you dial 911 as an AI or Cyborg!")
+				return
+			message_admins("[ADMIN_LOOKUPFLW(usr)] is considering calling the Sol Federation Marshals.")
+			if(tgui_input_list(usr, "Are you sure you want to call 911? Faulty 911 calls results in a $20,000 fine and a 5 year superjail \
+			sentence.", "Call 911", list("Yes", "No")) != "Yes")
+				return
+			message_admins("[ADMIN_LOOKUPFLW(usr)] has acknowledged the faulty 911 call consequences.")
+			if(tgui_input_list(usr, "You SHOULD call Marshals for:\n\
+			Security ignoring Command, Security violating civil rights, Security engaging in Mutiny, General Violation of Sol Federation Citizen Rights by Command/Security, etc.\n\
+			You SHOULD NOT call Marshals for:\n\
+			Corporate affairs, antagonist hunting, settling arguments, etc.\n\
+			Are you sure you want to call Marshals?", "Call Marshals", list("Yes", "No")) != "Yes")
+				return
+			message_admins("[ADMIN_LOOKUPFLW(usr)] has read and acknowleged the recommendations for what to call and not call Marshals for.")
+			var/reason_to_call_911 = stripped_input(usr, "What do you wish to call the SolFed Marshals for?", "Call 911", null, MAX_MESSAGE_LEN)
+			if(!reason_to_call_911)
+				to_chat(usr, "You decide not to call 911.")
+				return
 			GLOB.cops_arrived = TRUE
-			send_in_the_fuzz(EMERGENCY_RESPONSE_POLICE)
+			GLOB.call_911_msg = reason_to_call_911
 			GLOB.caller_of_911 = usr.name
-			to_chat(usr, span_notice("Authorization confirmed. 911 call dispatched to the Sol Federation Police Department."))
-			playsound(src, 'sound/machines/terminal_prompt_confirm.ogg', 50, FALSE)
+			log_game("[key_name(usr)] has called the Sol Federation Marshals for the following reason:\n[GLOB.call_911_msg]")
+			message_admins("[ADMIN_LOOKUPFLW(usr)] has called the Sol Federation Marshals for the following reason:\n[GLOB.call_911_msg]")
+			deadchat_broadcast(" has called the Sol Federation Marshals for the following reason:\n[GLOB.call_911_msg]", span_name("[usr.real_name]"), usr, message_type=DEADCHAT_ANNOUNCEMENT)
 
-			log_game("[key_name(usr)] has called the Sol Federation Police Department.")
-			message_admins("[ADMIN_LOOKUPFLW(usr)] has called the Sol Federation Police Department.")
-			deadchat_broadcast(" has called the Sol Federation Police Department.", span_name("[usr.real_name]"), usr, message_type=DEADCHAT_ANNOUNCEMENT)
+			send_in_the_fuzz(EMERGENCY_RESPONSE_POLICE)
```

Comment:
I'd still like a better name for this proc. It might mean something to you, but I don't know what a "fuzz" is and that's never something I'd search for.

Reply:
Sure.